syntax = "proto3";
package group;

message GroupObjectResponse {
  string group_id = 1;
  string group_name = 2;
  string group_avatar = 3;
  string group_type = 4;
  string created_by_client_id = 5;
  int64 created_at = 6;
  string updated_by_client_id = 7;
  repeated string lst_client_id = 8;
  int64 updated_at = 9;
}

message BaseResponse {
    bool success = 1;
}

// Request: create group
message CreateGroupRequest {
  string group_name = 1;
  string group_type = 2;
  string created_by_client_id = 3;
  repeated string lst_client_id = 4;
}
// Response: groupObjectResponse


// Request: update group
message UpdateGroupRequest {
  string group_id = 1;
  string group_name = 2;
  string group_avatar = 3;
  string updated_by_client_id = 4;
}
// Response: groupObjectResponse


// Request: get group
message GetGroupRequest {
  string group_id = 1;
}
// Response: groupObjectResponse


// Request: get joined groups
message GetJoinedGroupsRequest {
  string client_id = 1;
}
// Response
message GetJoinedGroupsResponse {
  repeated GroupObjectResponse lst_group = 1;
}

// Request: search groups
message SearchGroupsRequest {
  string keyword = 1;
}
// Response
message SearchGroupsResponse {
  repeated GroupObjectResponse lst_group = 1;
}

// Request: invite to group
message InviteToGroupRequest {
  string from_client_id = 1;
  string client_id = 2;
  string group_id = 3;
}
// Response BaseResponse


// Request: join group
message JoinGroupRequest {
  string client_id = 1;
  string group_id = 2;
}
// Response: BaseResponse


// Method
service Group {
    rpc create_group(CreateGroupRequest) returns (GroupObjectResponse) {};
    rpc get_group(GetGroupRequest) returns (GroupObjectResponse) {};
    rpc search_groups(SearchGroupsRequest) returns (SearchGroupsResponse) {};
    rpc get_joined_groups(GetJoinedGroupsRequest) returns (GetJoinedGroupsResponse) {};
    rpc invite_to_group(InviteToGroupRequest) returns (BaseResponse) {};
    rpc join_group(JoinGroupRequest) returns (BaseResponse) {};
}
